docs/tracking-events.md

```diff
@@ -238,12 +238,16 @@ class ProductDetailPage extends Component {
     this.selectVariant(product.variants[0]);
   }
 
-  // expects the prop `product`, with `variant` and `optionId` as function params
-  @trackProductViewed()
   selectVariant(variant, optionId) {
     // Do something with selected variant / option
+
+    this.trackAction({ variant, optionId, action: "Product Viewed" });
   }
 
+  // Expects the prop `product`, and an object with the following keys `variant`, `optionId`, and `action` as a function arg. The `product` prop is provided by the `ProductDetailPage` component.
+  @trackProduct()
+  trackAction(functionArgs) {}
```

Comment:
See below comment about maybe not needing the `functionArgs` param. If you don't need to use it then it is best not to include it.